packages/image-transform/lib/transform.js

```diff
@@ -107,7 +107,10 @@ const unsafeResizeFromBuffer = async (originalBuffer, options = {}) => {
         // CASE: Automatically remove metadata and rotate based on the orientation.
         .rotate();
 
-    if (options.format) {
+    const metadata = await s.metadata();
+    if (metadata.format === 'jpeg') {
+        s.jpeg({mozjpeg: true}); // .jpeg sets format
+    } else if (options.format) {
         s = s.toFormat(options.format);
```

Comment:
I'm not sure the logic here is quite right. Unless I'm misreading we're calling `.jpeg()` only when the original file is a jpeg but I think we want to call it in two cases?

1. when we're transforming a jpeg file
2. when we're transforming a different format to a jpeg file

The current logic appears to miss case 2 and also prevents a jpeg file from being transformed to a different format